src/tests/select-test/select-test.c

```diff
@@ -0,0 +1,539 @@
+
+    /* Connect to a server */
+    actual = OS_SocketConnect(c_socket_id, &s_addr, 10);
+    UtAssert_True(actual == expected, "OS_SocketConnect() (%ld) == OS_SUCCESS", (long)actual);
+
+    status = OS_TaskCreate(&s2_task_id, "Server2", Server_Fn2, OSAL_TASK_STACK_ALLOCATE, OSAL_SIZE_C(16384),
+                           OSAL_PRIORITY_C(50), 0);
+    UtAssert_True(status == OS_SUCCESS, "OS_TaskCreate() (%ld) == OS_SUCCESS", (long)status);
+
+    /* Connect to a server */
+    actual = OS_SocketConnect(c2_socket_id, &s2_addr, 10);
+    UtAssert_True(actual == expected, "OS_SocketConnect() (%ld) == OS_SUCCESS", (long)actual);
+
+    OS_SelectFdAdd(&ReadSet, c_socket_id);
+    OS_SelectFdAdd(&ReadSet, c2_socket_id);
+
+    UtAssert_True(OS_SelectFdIsSet(&ReadSet, c_socket_id), "OS_SelectFdIsSet(1) == true");
+    UtAssert_True(OS_SelectFdIsSet(&ReadSet, c2_socket_id), "OS_SelectFdIsSet(1) == true");
+
+    actual = OS_SelectMultiple(&ReadSet, &WriteSet, 100);
+    /* Verify Outputs */
+    UtAssert_True(actual == expected, "OS_SelectMultiple() (%ld) == OS_SUCCESS", (long)actual);
+
+    UtAssert_True(!OS_SelectFdIsSet(&ReadSet, c_socket_id), "OS_SelectFdIsSet(1) == false");
+    UtAssert_True(OS_SelectFdIsSet(&ReadSet, c2_socket_id), "OS_SelectFdIsSet(2) == true");
+
+    OS_BinSemFlush(bin_sem_id);
+    OS_BinSemDelete(bin_sem_id);
+}
+
+void TestSelectSingleWrite(void){
+    /*
+     * Test Case For:
+     * int32 OS_SelectSingle(uint32 objid, uint32 *StateFlags, int32 msecs);
+     */
+    int32  expected;
+    int32  actual;    
+
+    Setup_Server();
+    Setup_Client();
+    BinSemSetup();
+
+    expected = OS_SUCCESS;
+
+    /*
+     * Create a server thread, and connect client from
+     * this thread to server thread and verify connection
+     */
+
+    /* Create a server task/thread */
+    int32 status = OS_TaskCreate(&s_task_id, "Server", Server_Fn, OSAL_TASK_STACK_ALLOCATE, OSAL_SIZE_C(16384),
+                           OSAL_PRIORITY_C(50), 0);
+    UtAssert_True(status == OS_SUCCESS, "OS_TaskCreate() (%ld) == OS_SUCCESS", (long)status);
+
+    /* Connect to a server */
+    actual = OS_SocketConnect(c_socket_id, &s_addr, 10);
+    UtAssert_True(actual == expected, "OS_SocketConnect() (%ld) == OS_SUCCESS", (long)actual);   
+
+    uint32 StateFlags;
+    int count = 0;
+    char Buf_send_c[16834] = {0};
+    expected = OS_ERROR_TIMEOUT;
+
+    while (actual != OS_ERROR_TIMEOUT && count < max_buffer_loop) 
+    {
+        strcpy(Buf_send_c, "16 KB buffer filler");
+        actual   = OS_TimedWrite(c_socket_id, Buf_send_c, sizeof(Buf_send_c), 10);
+
+        StateFlags = OS_STREAM_STATE_WRITABLE;
+        actual = OS_SelectSingle(c_socket_id, &StateFlags, 100);        
+
+        count++;
+    }  
+
+    status = OS_BinSemGive(bin_sem_id);
+
+    if(count >= max_buffer_loop){
+         UtPrintf("Buffer size too large to test Select Single OS_STREAM_STATE_WRITABLE timeout\n");
```

Comment:
Suggest - "Unable to cause OS_STREAM_STATE_WRITEABLE timeout with large looped writes, skipping verification" and mark this as requires analysis.